Initialize scheduler's podInformer in sharedInformerFactory

Huang-Wei opened the pull request:
**What type of PR is this?**

/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

Scheduler used to initialize its podInfomer with specific filtering conditions: phase != Succeeded && phase != Failed. However, this caused a fragmentation with other informers - which are managed and initialized from the sharedInformerFactory.

Per https://github.com/kubernetes/kubernetes/pull/92801#discussion_r453553536, this PR uses `InformerFor` to initialize the podInfomer in-place.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

Some integration tests just initialize the podInfomer using `informerFactory.Core().V1().Pods()`, so they are not impacted. If an integration test wants to use a totally different podInformer, usually that needs a separate informerFactory.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

Huang-Wei commented:
/test pull-kubernetes-bazel-test

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
/aasign @ahg-g 
This PR is ready for review.

Huang-Wei commented:
/assign @ahg-g

chendave commented on pkg/scheduler/util/utils.go:
```diff
@@ -163,3 +166,20 @@ func ClearNominatedNodeName(cs kubernetes.Interface, pods ...*v1.Pod) utilerrors
 }
+
+// NewInformerFactory creates a SharedInformerFactory and initializes a scheduler specific
+// in-place podInformer.
```
I guess this is open to other `Informer` beside `podInformer`?

Huang-Wei replied:
Let's see. But now it just sticks to scheduler specific podInformer.

Huang-Wei commented:
Kindly ping @ahg-g, but it's not an urgent review.

ahg-g commented:
> Kindly ping @ahg-g, but it's not an urgent review.

we are in code freeze, no?

Huang-Wei commented:
> we are in code freeze, no?

Yes, but once code freeze is uplifted, code can be merged to 1.20.

ahg-g commented on pkg/scheduler/util/utils.go:
```diff
@@ -163,3 +166,20 @@ func ClearNominatedNodeName(cs kubernetes.Interface, pods ...*v1.Pod) utilerrors
+
+// NewInformerFactory creates a SharedInformerFactory and initializes a scheduler specific
+// in-place podInformer.
+func NewInformerFactory(cs kubernetes.Interface, resyncPeriod time.Duration) informers.SharedInformerFactory {
```
can we keep this in the scheduler pkg? we are close to having a staging repo for common utility functions, so this pkg will very likely be removed in the 1.20.

Huang-Wei replied:
Sure. Updated.

ahg-g reviewed (commented):
one nit

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

ahg-g commented:
Thanks, can you please squash.

Huang-Wei commented:
> Thanks, can you please squash.

Thanks. Updated.

ahg-g commented:
/lgtm

Huang-Wei commented:
/retest

Huang-Wei commented:
/retest

Huang-Wei commented:
@oomichi would you mind taking a look at the changes at `test` pkg?

oomichi reviewed (commented):
/test part seems good for me, that is straightforward for corresponding /pkg change.

/approve

oomichi commented:
> @oomichi: The specified target(s) for /test were not found.

Oops..

Huang-Wei commented:
@oomichi Thanks!

Huang-Wei commented:
/retest

Huang-Wei commented:
A side note: this PR would conflict with #93566, and they may be merged in the same phase (step 3) per https://docs.google.com/document/d/1yI7o7R2bOB7QtkHUFMHMb5XObiJmzIAnJPuxhdAX-CA/edit#. So which one gets merged will block the merge of the other. And I hope these 2 PRs would not block the whole merge queue.

Huang-Wei commented:
/test pull-kubernetes-e2e-kind-ipv6

Huang-Wei commented:
/milestone v1.20

Huang-Wei committed:
Initialize scheduler's podInformer in sharedInformerFactory

Scheduler's specific podInfomer is now initialized inside the sahredInformerFactory.

Huang-Wei commented:
@ahg-g needs another /lgtm - fixed a signature mismatch in an integration test.

ahg-g commented:
/lgtm

Huang-Wei commented:
/retest
